### PR TITLE
run core acceptance tests on LDAP using drone

### PR DESCRIPTION
**Pull request by individual-it**
run core UI tests that are marked with `@TestAlsoOnExternalUserBackend` 
users/groups will not be created on the fly but pulled in from an ldif file

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    commands:
+      - cd /var/www
+      - wget -q http://78.47.62.142/archisdownloads/owncloud-daily-master-qa-patched.tar.bz2
+      - tar -xjf owncloud-daily-master-qa-patched.tar.bz2
+      - cd /var/www/owncloud/
+      - php occ maintenance:install -vvv --database=sqlite --database-name=owncloud --database-table-prefix=oc_ --admin-user=admin --admin-pass=admin --data-dir=/var/www/owncloud/data
+      - cd /var/www/owncloud/apps/user_ldap
+      - make
+      - cd /var/www/owncloud/
+      - php occ a:l
```
Comment (phil-davis) [inline .drone.yml]: Everyone seems to use shorthand for ``app:list`` ``app:enable`` ...

File: lib/Command/SetConfig.php [hunk 2]
```diff
@@ -60,6 +60,7 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 		$availableConfigs = $helper->getServerConfigurationPrefixes();
 		$configID = $input->getArgument('configID');
 		if(!in_array($configID, $availableConfigs)) {
+			var_dump($availableConfigs);
```
Comment (phil-davis) [inline lib/Command/SetConfig.php]: bonus debugging code

Comment (individual-it) [reply lib/Command/SetConfig.php]: will have to be cleaned up before merging, would like to keep it while [WIP]

File: .drone.yml (hunk 1 above)
Comment (patrickjahns) [reply .drone.yml]: Regarding install - can you switch to using https://github.com/owncloud-ci/core

It allows for using a branch/tarball

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
+workspace:
+  base: /var/www/owncloud
+  path: apps/user_ldap
+
+pipeline:
+  ldap:
+    image: dinkel/openldap
```
Comment (patrickjahns) [inline .drone.yml]: What made you use this ldap container - compared to `osixia/openldap` which we use in other infrastructure projects?

Comment (individual-it) [reply .drone.yml]: this script https://github.com/owncloud/administration/blob/master/ldap-testing/start.sh

Comment (patrickjahns) [reply .drone.yml]: Ah didn't find these before :'(

**Comment by individual-it** [comment]
@phil-davis @patrickjahns I feel this is ready to go. Any more comments/change requests or should we merge?

File: lib/Command/SetConfig.php (hunk 2 above)
Comment (phil-davis) [reply lib/Command/SetConfig.php]: Debug line is now gone 👍

**Comment by patrickjahns** [comment]
What's the status on the core side? If we have the required changes in master/stable10 please update the `install_server` step to use nightly tarballs again.

It looks fine for the first iteration - I'd like to also port the tests suites from .travis to drone at some point

**Comment by individual-it** [comment]
thank you for reminding to change the source.
core should be fine, but looks like something changed in master in between.

cannot sync users
```
An unhandled exception has been thrown:
TypeError: Argument 1 passed to OC\User\SyncService::syncAccount() must be an instance of OC\User\Account, null given, called in /var/www/owncloud/lib/private/User/SyncService.php on line 121 and defined in /var/www/owncloud/lib/private/User/SyncService.php:279
Stack trace:
#0 /var/www/owncloud/lib/private/User/SyncService.php(121): OC\User\SyncService->syncAccount(NULL, Object(OCA\User_LDAP\User_Proxy))
#1 /var/www/owncloud/core/Command/User/SyncBackend.php(148): OC\User\SyncService->run(Object(OCA\User_LDAP\User_Proxy), Object(Closure))
#2 /var/www/owncloud/lib/composer/symfony/console/Command/Command.php(252): OC\Core\Command\User\SyncBackend->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#3 /var/www/owncloud/lib/composer/symfony/console/Application.php(938): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 /var/www/owncloud/lib/composer/symfony/console/Application.php(240): Symfony\Component\Console\Application->doRunCommand(Object(OC\Core\Command\User\SyncBackend), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 /var/www/owncloud/lib/composer/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 /var/www/owncloud/lib/private/Console/Application.php(161): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /var/www/owncloud/console.php(106): OC\Console\Application->run()
#8 /var/www/owncloud/occ(11): require_once('/var/www/ownclo...')
    1 [->--------------------------] 0/6 [>---------------------------]   0%
#9 {main}
```

I will look into it

**Comment by individual-it** [comment]
works fine on stable10 but on master there is an issue see https://github.com/owncloud/core/issues/30369

**Comment by tomneedham** [comment]
Restart CI - master fix PR is in.

**Comment by patrickjahns** [comment]
Restarted builds - when green we can merge

**Comment by individual-it** [comment]
CI will not become green because it using the daily builds

**Comment by individual-it** [comment]
https://github.com/owncloud/core/pull/30367 fixes the issue, CI should be green tomorrow

**Pull request merged**